csource/csource_test.go

```diff
@@ -70,11 +92,37 @@ func TestSyz(t *testing.T) {
 	testOne(t, p, opts)
 }
 
-func Test(t *testing.T) {
+func TestOptionsSingle(t *testing.T) {
+	rs, _ := initTest(t)
+	syzProg := prog.GenerateAllSyzProg(rs)
+	t.Logf("syz program:\n%s\n", syzProg.Serialize())
+	for i, opts := range allOptionsSingle() {
+		t.Run(fmt.Sprintf("%v", i), func(t *testing.T) {
+			rs, iters := initTest(t)
+			t.Logf("opts: %+v", opts)
+			for i := 0; i < iters; i++ {
+				p := prog.Generate(rs, 10, nil)
+				testOne(t, p, opts)
+			}
+			testOne(t, syzProg, opts)
+		})
+	}
+}
+
+func TestOptionsPermutations(t *testing.T) {
 	rs, _ := initTest(t)
 	syzProg := prog.GenerateAllSyzProg(rs)
 	t.Logf("syz program:\n%s\n", syzProg.Serialize())
-	for i, opts := range allOptionsPermutations() {
+	permutations := allOptionsPermutations()
```

Comment:
I think it makes sense to merge TestOptionsSingle and TestOptionsPermutations because there is considerable code duplication. Namely, fill in permutations array depending on short/not-short and then have common code. Will also reduce non-short test time as currently we test all permutations and then also all options independently.